docs/tools_config/memory-object.md

```diff
@@ -0,0 +1,138 @@
+---
+sidebar_position: 3
+---
+
+# Memory Object
+## Introduction
+A memory object serve the purpose of transferring data between multiple 
+independent tools within one conversation. This is useful when you want to keep 
+important contextual information. Think of it as like a train of thought.
+## How to use:
+When writing a tool add in a memory object within the tool parameters and then return.
+You can return it just by itself
+```python
+def introductions(user_name: str, memory: dict):
+    memory["user_name"] = user_name
+    return memory
```

Comment:
this doesn't work, when you amke changes to the memory object

you should ALWAYS return an object with the keys

`responseString` and `memory`